pkg/vsphere/vm/vm.go

```diff
@@ -116,3 +116,34 @@ func (vm *VirtualMachine) FetchExtraConfig(ctx context.Context) (map[string]stri
 	}
 	return info, nil
 }
+
+// WaitForExtraConfig waits until key shows up with the expected value inside the ExtraConfig
+func (vm *VirtualMachine) WaitForExtraConfig(ctx context.Context, key string, expected string) error {
```

Comment:
This should take a function to do the validation rather than an expected string.
This currently hangs indefinitely in the error case - the Start call needs to block until this is non-nil, then return success or error based on the contents.